test/read-stream.js

```diff
@@ -54,6 +54,88 @@ describe( 'BlockMap.ReadStream', function() {
 
   })
 
+  it( 'should not close the fd if autoClose = false', function( done ) {
+
+    var filename = path.join( __dirname, '/data/bmap.img' )
+    var blockMap = BlockMap.create( require( './data/version-2.0' ) )
+    var fd = fs.openSync( filename, 'r' )
+    var blockCount = 0
+    var readStream = new BlockMap.ReadStream( null, blockMap, {
+      fd: fd,
+      autoClose: false,
+    })
+
+    readStream
+      .on( 'data', ( block ) => {
+        blockCount++
+        assert.ok( block.address != null, 'block address missing' )
+        assert.ok( block.position != null, 'block position missing' )
+      })
+      .once( 'error', done )
+      .once( 'end', function() {
+        assert.equal( this.blocksRead, blockMap.mappedBlockCount, 'blocksRead mismatch' )
+        assert.equal( this.bytesRead, blockMap.mappedBlockCount * blockMap.blockSize, 'bytesRead mismatch' )
+        assert.equal( this.rangesRead, blockMap.ranges.length, 'rangesRead mismatch' )
+        assert.equal( blockCount, blockMap.mappedBlockCount, 'actual blocks read mismatch' )
+        assert.ok( fs.fstatSync( readStream.fd ) )
+        fs.closeSync( readStream.fd )
+        done()
+      })
+
+  })
+
+  it( 'should start reading at `start`, if specified', function( done ) {
+
+    var filename = path.join( __dirname, '/data/padded-bmap.img' )
+    var blockMap = BlockMap.create( require( './data/version-2.0' ) )
+    var blockCount = 0
+    var readStream = new BlockMap.ReadStream( filename, blockMap, {
+      start: 4096,
+    })
+
+    readStream
+      .on( 'data', ( block ) => {
+        blockCount++
+        assert.ok( block.address != null, 'block address missing' )
+        assert.ok( block.position != null, 'block position missing' )
+      })
+      .once( 'error', done )
+      .once( 'end', function() {
+        assert.equal( this.blocksRead, blockMap.mappedBlockCount, 'blocksRead mismatch' )
+        assert.equal( this.bytesRead, blockMap.mappedBlockCount * blockMap.blockSize, 'bytesRead mismatch' )
+        assert.equal( this.rangesRead, blockMap.ranges.length, 'rangesRead mismatch' )
+        assert.equal( blockCount, blockMap.mappedBlockCount, 'actual blocks read mismatch' )
```

Comment:
This is node core fs streams' default behavior, I don't think that needs explicit mentioning.

> Perhaps this means the unit-tests should be using padding-start.img, padding-end.img and padding-both.img files?

Not sure I'm following. As it is, all three cases can be (and are) tested with one padded image. Appending more bytes wouldn't do anything, as it will only read the mapped ranges anyways (which is tested in plenty of other test already written).
Also, by not padding the end we can detect errors where an applied `start` offset would make it read past the actual end.